frontend/src/components/Division.tsx

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import { IUser } from '../common/interfaces/user.interface';
+import i18n from '../common/config/i18n';
+import { formatDate } from '../common/utils/utils';
+import Card from '../layouts/CardLayout';
+import CardHeader from './CardHeader';
+import CardBody from './CardBody';
+import {
+  ChevronUpDownIcon,
+  EyeIcon,
+  PencilIcon,
+  PlusIcon,
+  TrashIcon,
+} from '@heroicons/react/24/outline';
+import Button from './Button';
+import DataTableComponent from './DataTableComponent';
+import { PaginationConfig } from '../common/constants/pagination';
+import EmptyContent from './EmptyContent';
+import LoadingContent from './LoadingContent';
+import { SortOrder, TableColumn } from 'react-data-table-component';
+import Popover from './Popover';
+import { IBaseEntity } from '../common/interfaces/base-entity.interface';
+import { IPaginatedEntity } from '../common/interfaces/paginated-entity.interface';
+import Tabs from './Tabs';
+import { SelectItem } from './Select';
+
+export enum DivisionType {
+  Branches = 'branches',
+  Departments = 'departments',
+  Roles = 'roles',
+}
+
+export interface IDivision extends IBaseEntity {
+  name: string;
+  createdBy: Pick<IUser, 'id' | 'name'>;
+  membersCount: number;
+  createdOn: Date | string;
+}
+
+export const DivisionTabs: SelectItem[] = [
+  { key: 0, value: i18n.t('division:branches') },
+  { key: 1, value: i18n.t('division:departments') },
+  { key: 2, value: i18n.t('division:roles') },
+];
+
+export const DivisionTableHeader = [
+  {
+    id: 'name',
+    name: i18n.t('general:name'),
+    sortable: true,
+    grow: 2,
+    minWidth: '15rem',
+    selector: (row: IDivision) => row.name,
+  },
+  {
+    id: 'membersCount',
+    name: i18n.t('division:members'),
+    sortable: true,
+    grow: 2,
+    minWidth: '15rem',
+    selector: (row: IDivision) => row.membersCount,
+  },
+  {
+    id: 'createdBy.name',
+    name: i18n.t('general:created_by'),
+    sortable: true,
+    grow: 2,
+    minWidth: '15rem',
+    selector: (row: IDivision) => row.createdBy.name,
+  },
+  {
+    id: 'createdOn',
+    name: i18n.t('general:created_on'),
+    sortable: true,
+    grow: 2,
+    minWidth: '15rem',
+    selector: (row: IDivision) => formatDate(row.createdOn),
+  },
+];
+
+interface DivisionProps {
+  data?: IPaginatedEntity<IDivision>;
+  isLoading: boolean;
+  divisionType: DivisionType;
+  page: number;
+  onSort: (column: TableColumn<IDivision>, direction: SortOrder) => void;
+  onChangePage: (newPage: number) => void;
+  onRowsPerPageChange: (rows: number) => void;
+  onTabChange: (id: number) => void;
+}
+
+const Division = ({
```

Comment:
If this incorporate the tabs as well we should call it Divisions as it refers to all of them. And we should also rename the file and prop interface.